src/OpenTelemetry.AutoInstrumentation/Configuration/MeterSettings.cs

```diff
@@ -0,0 +1,159 @@
+            if (providerPlugins != null)
+            {
+                foreach (var pluginAssemblyQualifiedName in providerPlugins.Split(':'))
+                {
+                    MetricPlugins.Add(pluginAssemblyQualifiedName);
+                }
+            }
+
+            var additionalSources = source.GetString(ConfigurationKeys.Metrics.AdditionalSources);
+            if (additionalSources != null)
+            {
+                foreach (var sourceName in additionalSources.Split(separator: ','))
+                {
+                    Meters.Add(sourceName);
+                }
+            }
+
+            MetricsEnabled = source.GetBool(ConfigurationKeys.Metrics.Enabled) ?? true;
+            LoadMetricsAtStartup = source.GetBool(ConfigurationKeys.Metrics.LoadMeterAtStartup) ?? true;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether metrics is enabled.
+        /// Default is <c>true</c>.
+        /// </summary>
+        /// <seealso cref="ConfigurationKeys.Metrics.Enabled"/>
+        public bool MetricsEnabled { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the metrics should be loaded by the profiler. Default is true.
+        /// </summary>
+        public bool LoadMetricsAtStartup { get; }
+
+        /// <summary>
+        /// Gets the metrics exporter.
+        /// </summary>
+        public MetricsExporter MetricExporter { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the console exporter is enabled.
+        /// </summary>
+        public bool ConsoleExporterEnabled { get; }
+
+        /// <summary>
+        /// Gets the list of plugins represented by <see cref="Type.AssemblyQualifiedName"/>.
+        /// </summary>
+        public IList<string> MetricPlugins { get; } = new List<string>();
+
+        /// <summary>
+        /// Gets the list of enabled meters.
+        /// </summary>
+        public IList<MeterInstrumentation> EnabledInstrumentation { get; }
+
+        /// <summary>
+        /// Gets the list of meters to be added to the MeterProvider at the startup.
+        /// </summary>
+        public IList<string> Meters { get; } = new List<string>();
+
+        internal static MeterSettings FromDefaultSources()
+        {
+            var configurationSource = new CompositeConfigurationSource
+            {
+                new EnvironmentConfigurationSource(),
+
+#if NETFRAMEWORK
+                // on .NET Framework only, also read from app.config/web.config
+                new NameValueConfigurationSource(System.Configuration.ConfigurationManager.AppSettings)
+#endif
+            };
+
+            return new MeterSettings(configurationSource);
+        }
+
+        private static MetricsExporter ParseMetricExporter(IConfigurationSource source)
+        {
+            var metricsExporterEnvVar = source.GetString(ConfigurationKeys.Metrics.Exporter) ?? "otlp";
+            switch (metricsExporterEnvVar)
```

Comment:
Should we do a `string.ToLower()` here just to be safe?

Reply:
Very good recommendation, we have this pattern uses in TracerSettings and other places. I want to get this covered in a different PR. What are your thoughts.

Reply:
Yes another PR is fine with me 😄